resource/config_map.go

```diff
@@ -0,0 +1,198 @@
+package resource
```

Comment:
Can we please have a `resource/configmap` package? I really do not like having so many different implementations in the same package. The naming is then also easier.

Reply:
That would make me very sad. This follows package per dependency principle. Dependency here is `client-go/kubernetes`. I don't think we should ever implement resources for other clients in operatorkit. I just case about code being readable and having nice APIs. `configmap.New` is just useless information while `resource.NewConfigMap` is meaningful. Naming isn't an issue at all here as we have single dependency. There are no object name collisions in Kubernetes.

I'd like to start with that. If this doesn't work we can deprecate the package and create subpackages.

Reply:
+1

Reply:
@tuommaki `+1` on what? :)

Reply:
God damn GItHub, comments out-of-date when reviewing...

So originally I was adding +1 for Tim's comment, but given your explanation, I agree with you and I'm fine with this packaging.

Reply:
That is a pretty bad argument IMO because fucking everything depends on `client-go/kubernetes` so we should then rather habe a mono repo.